CONTRIBUTING.md

```diff
@@ -1,24 +1,83 @@
----
-name: General Guidelines
-route: /contribute/guidelines
-menu: Contribute
----
-
 # Contributing
 
 In the first place, thank you for your interest in contributing! 🙏
 
 ## Development
 
-To build the project:
+Working on the site requires:
+
+* [Docker]
+* [Docker Compose]
+
+This allows running the documentation site which serves as a development platform.
+
+### Configure Docker Compose
+
+Review the default env variable values in the `docker-compose.yml` file.
+The defaults should work for most systems, but can be changed if needed.
+
+To change them:
+
+1. Create a `.env` file from the template:
+
+    ```bash
+    cp .env.dist .env
+    ```
+
+2. Edit the new `.env` file as needed
+
+### Use Docker Compose
+
+#### Node shell
+
+All npm commands such as `npm ci`, `npm test`, `npm run eslint` and others you
+need to run them within the `node_shell` docker container.
+
+To log into the container, run:
+
+```bash
+docker compose run --rm node_shell
+```
+
+#### Run the Dev Server
+
+1. Install dependencies (run within `node_shell`):
+
+    ```bash
+    npm ci
+    ```
+
+2. Run development server (run on host):
+
+    ```bash
+    docker compose up node_dev_server mkdocs_dev_server
+    ```
+
+#### Build the Project
+
+1. Make sure the dev server is not running (run on host):
+
+    ```bash
+    docker compose down
+    ```
 
-1. Install local npm packages: `npm install`
-2. Run the build: `npm run build`
+2. Install dependencies (run within `node_shell`):
 
-To run the dev server:
+    ```bash
+    npm ci
+    ```
 
-1. Install local npm packages: `npm install`
-2. Run the dev server (usually on `http://localhost:3000`): `npm start`
+3. Build JS (run on host):
+
+    ```bash
+    docker compose run --rm node_build_site
+    ```
+
+4. Build mkDocs (run on host):
+
+    ```bash
+    docker compose run --rm mkdocs_build_site
+    ```
 
 ## Git Workflow
```

Comment:
The nested list here is rendered flat and numbered:

<img width="985" alt="obrazek" src="https://github.com/react-ui-org/react-ui/assets/5614085/8f34090f-ae05-48e4-bdde-4174b5e8d06f">